### ISSUE TITLE
Contiguous IDL link problems with case-insensitive-matching IDL elements

### ISSUE THREAD

**Issue by mwatson2** [w3c/respec#893]
Links are incorrect when two IDL elements are a case-insensitive match.

For example, in [media source extensions](https://w3c.github.io/media-source/#mediasource) (see [this PR ](https://github.com/w3c/media-source/pull/127) for Contiguous IDL version), we have an attribute named `readyState` of type `ReadyState`, where the type is defined in the same document.

I expect `readyState` to link to `#dom-readystate` and `ReadyState` to link to `#idl-def-readystate`, but what happens is that `ReadyState` links to `#dom-readystate` and `readyState` does not have a link at all.

**Comment by mwatson2** [comment]
@marcoscaceres We are hoping to publish the Media Source Extensions next week and would like to get our migration to Contiguous IDL included.

Could you give me an ETA for this bug ? Or, some pointers as to what needs to be done to fix it so I could fix it myself ?

**Comment by marcoscaceres** [comment]
@mwatson2, apologies, I'm at a conference this week, so it's unlikely that I will be able to get to this at least this week. There is "webidl-contiguous.js" file in "js/core/" that handles most of this. You could poke around in there.

**Comment by ddorwin** [comment]
As noted in https://github.com/w3c/encrypted-media/issues/293#issuecomment-238312853, all references to the ObjectType (`<a>ObjectType</a>`) in prose become links to the domAttribute definition, and references to the ObjectType (`<a>ObjectType</a>`) in definitions of IDL items become links to the domAttribute in the actual IDL.

**Comment by ddorwin** [comment]
For the instance of this reported in w3c/encrypted-media#293, the `definitionMap` in the ReSpec output contains:
```
        "mediakeys": [
          "dom-mediakeys",
          "idl-def-mediakeys",
          "idl-def-htmlmediaelement-mediakeys"
        ],```

It appears something has merged the two types into a single definition. This is probably an artifact and not a cause, since both the first and last entries are used.

**Comment by mfoltzgoogle** [comment]
In the Presentation API [1], we have a slightly different case, with an idiom `presentation` (defined in prose) and an interface `Presentation` (defined in IDL).  We're manually linking the former to work around this issue (see https://github.com/w3c/presentation-api/pull/403).

[1] https://w3c.github.io/presentation-api/

**Comment by marcoscaceres** [comment]
closed via  #2324
